policy/scan/pdque/queue_test.go

```diff
@@ -0,0 +1,407 @@
+	testDir, err := os.MkdirTemp("", "test_queue")
+	if err != nil {
+		t.Fatalf("Failed to create temp dir: %v", err)
+	}
+	defer os.RemoveAll(testDir)
+
+	// Instantiate the Queue.
+	queue, err := New("testQueue", testDir, 1000, func() interface{} {
+		return nil
+	})
+	if err != nil {
+		t.Fatalf("Failed to create queue: %v", err)
+	}
+
+	// Define a test job (could be any struct or type that you will be enqueueing).
+	testJob := struct {
+		Data string
+	}{
+		Data: "test data",
+	}
+
+	// Enqueue the job.
+	err = queue.Enqueue([]byte(fmt.Sprintf("%v", testJob)))
+	if err != nil {
+		t.Errorf("Failed to enqueue job: %v", err)
+	}
+
+	// Verify that a file has been created in the queue directory.
+	files, err := os.ReadDir(testDir)
+	if err != nil {
+		t.Fatalf("Failed to read queue directory: %v", err)
+	}
+
+	if len(files) != 1 {
+		t.Fatalf("Expected 1 file in queue directory, found %d", len(files))
+	}
+}
+
+func TestClose(t *testing.T) {
+	testDir, err := os.MkdirTemp("", "test_queue")
+	if err != nil {
+		t.Fatalf("Failed to create temp dir: %v", err)
+	}
+	defer os.RemoveAll(testDir)
+
+	// Create some temporary files to simulate the state of the queue with pending jobs.
+	tempFiles := []string{".tmp1", ".tmp2", ".tmp3"}
+	for _, f := range tempFiles {
+		tmpFilePath := filepath.Join(testDir, f)
+		if err := os.WriteFile(tmpFilePath, []byte("data"), 0o644); err != nil {
+			t.Fatalf("Failed to create temp file: %v", err)
+		}
+	}
+
+	// Instantiate the Queue.
+	queue, err := New("testQueue", testDir, 1000, func() interface{} {
+		return nil
+	})
+	if err != nil {
+		t.Fatalf("Failed to create queue: %v", err)
+	}
+
+	// Close the queue.
+	if err := queue.Close(); err != nil {
+		t.Errorf("Failed to close queue: %v", err)
+	}
+
+	// Verify that the queue is marked as closed.
+	if !queue.closed {
+		t.Errorf("Queue should be marked as closed.")
+	}
+
+	// Verify that temporary files are cleaned up.
+	files, err := os.ReadDir(testDir)
+	if err != nil {
+		t.Fatalf("Failed to read queue directory: %v", err)
+	}
+	for _, file := range files {
+		if strings.HasPrefix(file.Name(), ".") {
+			t.Errorf("Temporary file %s was not cleaned up", file.Name())
+		}
+	}
+
+	// Verify that no new actions can be performed on the queue.
+	if err := queue.Enqueue([]byte("data")); err == nil {
+		t.Errorf("Enqueue should fail on a closed queue")
+	}
+
+	// Verify that the Dequeue method also behaves as expected.
+	if _, err := queue.Dequeue(); err == nil {
+		t.Errorf("Dequeue should fail on a closed queue")
+	}
+}
+
+type testObj struct {
+	Name string
+	ID   int
+}
+
+func TestEnqueueDequeue(t *testing.T) {
+	testDir, err := os.MkdirTemp("", "test_enqueue_dequeue")
+	if err != nil {
+		t.Fatalf("Failed to create temp dir: %v", err)
+	}
+	defer os.RemoveAll(testDir)
+	// Setup: Initialize the queue
+	queue, err := NewOrOpen("testQueue", testDir, 10, func() interface{} {
+		return new(testObj)
+	})
+	if err != nil {
```

Comment:
rework this checks with `require.NoError(err)`. Makes the code much shorter and easier to read